cassovary-core/src/main/scala/com/twitter/cassovary/algorithms/similarity/Similarity.scala

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2015 Twitter, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package com.twitter.cassovary.algorithms.similarity
+
+import com.twitter.cassovary.graph.{DirectedGraph, Node}
+import com.twitter.cassovary.graph.GraphDir.GraphDir
+
+case class SimilarNode(nodeId: Int, similarityScore: Double)
+
+trait Similarity {
+
+  def graph: DirectedGraph[Node]
+
+  protected def getNeighbors(dir: GraphDir, nodeId: Int) = graph.getNodeById(nodeId).get.neighborIds(dir).toSet
+
+  def calculateSimilarity(dir: GraphDir, u: Int, v: Int): Double
+
+  def getTopKSimilarNodes(dir: GraphDir, u: Int, k: Int): Seq[Int] = {
+    val similarNodesSeq = graph.foldLeft(Seq.empty[SimilarNode]) { (partialSeq, node) =>
+      partialSeq ++ (
+        if (node.id != u) Seq(new SimilarNode(node.id, calculateSimilarity(dir, u, node.id)))
+        else Seq.empty[SimilarNode]
+        )
+    }
+    similarNodesSeq.sortBy(node => -node.similarityScore).map(node => node.nodeId).take(k)
+  }
```

Comment:
But, we have to find the neighbors of `U` to find common neighbors. 
After that, It takes constant time to get the size of neighbors of `U`